File: output/xCodeEval_remove_8000/C++/Go/40ca10a54463a16f6c0559d2d974fdc5.go
package main

import (
	"fmt"
	"math"
)

const N = 18000
const M = 18000

var is [N]bool
var prm [M]int
var cntp int

func id0(n int) int {
	k := 0
	s := int(math.Sqrt(float64(n)) + 1)
	for i := 0; i < n; i++ {
		is[i] = true
	}
	prm[k] = 2
	is[0] = false
	is[1] = false
	k++
	for i := 4; i < n; i += 2 {
		is[i] = false
	}
	for i := 3; i < s; i += 2 {
		if is[i] {
			prm[k] = i
			k++
			for j, s := i*2, i*i; j < n; j += s {
				is[j] = false
			}
		}
	}
	for i := s; i < n; i += 2 {
		if is[i] {
			prm[k] = i
			k++
		}
	}
	return k
}

File: output/xCodeEval_remove_8000/C++/Java/40ca10a54463a16f6c0559d2d974fdc5.java
import java.util.Arrays;

public class 40ca10a54463a16f6c0559d2d974fdc5xcodeeval_processed_8000.json {
    static final int N = 18000;
    static final int M = 18000;
    static boolean[] is = new boolean[N];
    static int[] prm = new int[M];
    static int cntp;

    static int id0(int n) {
        int k = 0;
        int s, e = (int) (Math.sqrt(0.0 + n) + 1);
        Arrays.fill(is, true);
        prm[k++] = 2;
        is[0] = is[1] = false;
        for (int i = 4; i < n; i += 2) is[i] = false;
        for (int i = 3; i < e; i += 2) {
            if (is[i]) {
                prm[k++] = i;
                for (s = i * 2, int j = i * i; j < n; j += s) {
                    is[j] = false;
                }
            }
        }
        for (int i = 0; i < n; i += 2) {
            if (is[i]) prm[k++] = i;
        }
        return k;
    }

